### PR TITLE
feat: finalize gcp self-hosted deployment example

### PR DESCRIPTION
**Pull request by abelanger5**
Moves the GCP example to the `self-hosted` directory, updates the README and makes it more of a "production-ready" deployment. In particular, uses the RabbitMQ operator with a cluster in HA mode and adds the `hatchet-ha` Helm chart instead of a custom chart.

### CODE REVIEW

File: sns-ecs-example/demo-app/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+# Install dependencies
 RUN pnpm install
-COPY index.ts .
 
-RUN npx tsc index.ts
+# Copy the rest of the source code
+COPY . .
+
+# Run TypeScript compiler explicitly specifying the config file
+RUN npx tsc --project tsconfig.json
```
Comment (grutt) [inline sns-ecs-example/demo-app/Dockerfile:17]: should we include a multi stage build here? cloud examples has an example Dockerfile

**Review by grutt — COMMENTED** [review]
@abelanger5 i'm seeing db conns are bumped to 1k, but did you increase engine and api connections anywhere? i didnt see this in the diff

**Review by grutt — APPROVED** [review]

**Comment by abelanger5** [comment]
> i'm seeing db conns are bumped to 1k, but did you increase engine and api connections anywhere? i didnt see this in the diff

@grutt yes: https://github.com/hatchet-dev/hatchet-infra-examples/blob/d7eadf003c757b3d26fd52c31162ecfed45bbc37/self-hosting/gcp/modules/hatchet/hatchet.tf#L263

File: sns-ecs-example/demo-app/Dockerfile (hunk 1 above)
Comment (abelanger5) [reply sns-ecs-example/demo-app/Dockerfile:17]: This was just to quickly fix an issue I noticed -- eventually this should be replaced with the local setup repo that we're working on

**Pull request merged**